pennylane/devices/qubit/apply_operation.py

```diff
@@ -258,3 +265,75 @@ def apply_snapshot(op: qml.Snapshot, state, is_state_batched: bool = False, debu
         else:
             debugger.snapshots[len(debugger.snapshots)] = flat_state
     return state
+
+
+# pylint:disable = no-value-for-parameter, import-outside-toplevel
+@apply_operation.register
+def apply_parametrized_evolution(
+    op: qml.pulse.ParametrizedEvolution, state, is_state_batched: bool = False, debugger=None
+):
+    """Apply ParametrizedEvolution by evolving the state rather than the operator matrix
+    if we are operating on more than half of the subsystem"""
+    # shape(state) is static (not a tracer), we can use an if statement
+    num_wires = len(qml.math.shape(state))
```

Comment:
does parametrized_evolution not support batching? there isn't any handling for it here, and you call `flatten()` explicitly below. if you do add support, you can flatten the state like [this](https://github.com/PennyLaneAI/pennylane/blob/cbc8d19e6a36f0d4d38b44eabd4a8b98db677327/pennylane/devices/qubit/measure.py#L63-L65)

Reply:
Supporting batching in `ParameterizedEvolution` is a thing that comes up once every few weeks, and then everyone looks at it and shrugs because its not clear how realistic that is, and then David says he'll try to look at it if its important to anyone and see if he can figure anything out, and then everyone moves on.

But can't hurt to add that already I guess, in case batching is added in the future.

Reply:
i'd instead just raise an error if `is_state_batched` is true and note it in a story somewhere for David :p

Reply:
Added an error but I want to run this by Korbinian on Monday and make sure he agrees that batching isn't something it should be able to do.

Reply:
Is there an update on the batching situation? I'm okay with raising an error, just want to know if any stakeholders care.

Reply:
It does not support batching/parameter broadcasting. The language in the gradient methods regarding 'broadcasting' is referring to the broadcasting over time that happens if `return_intermediate=True`, which is a feature of the jax ODE solver, and unrelated to how things are handled in PennyLane.

Reply:
Should we run this check in preprocessing? If any parametrized evolutions exist in the circuit, we use `broadcast_expand`?

Reply:
I think currently its not very important to anyone

Reply:
added [a story](https://app.shortcut.com/xanaduai/story/45833/support-batching-with-parametrizedevolution) to the backlog to support batching in the future